packages/create-pwa/lib/index.js

```diff
@@ -1,17 +1,38 @@
 const { basename, resolve } = require('path');
 const os = require('os');
+const fetch = require('node-fetch');
 const changeCase = require('change-case');
 const inquirer = require('inquirer');
 const execa = require('execa');
 const chalk = require('chalk');
 const gitUserInfo = require('git-user-info');
 const isInvalidPath = require('is-invalid-path');
 const isValidNpmName = require('is-valid-npm-name');
+const { uniqBy } = require('lodash');
+
 const pkg = require('../package.json');
 const {
-    sampleBackends
+    sampleBackends: defaultSampleBackends
 } = require('@magento/pwa-buildpack/lib/cli/create-project');
 
+const removeDuplicateBackends = backendEnvironments =>
+    uniqBy(backendEnvironments, 'url');
+
+const fetchSampleBackends = async () => {
+    try {
+        const res = await fetch(
+            'https://fvp0esmt8f.execute-api.us-east-1.amazonaws.com/default/getSampleBackends'
```

Comment:
I am looking at AWS docs to use a DNS name instead of this raw path. It will be changed in the future (before this PR is merged).

Reply:
Is this a lambda? Does it read configuration somewhere? I'm a bit surprised we don't just maintain a list in the extension.

Reply:
The plan is to pull these backend URLs dynamically so we don't need to publish every single time a backend changes, which we will have to if these are part of the package. The URL above is an AWS Gateway URL. The plan is to use a domain-specific URL so in the future if we move between cloud providers (Azure, Ethos, etc), we can keep using the same URL but the implementation in the back can change.

Reply:
> so we don't need to publish every single time a backend changes

_Something_ will have to change _somewhere_ whether that's a file in s3, a lambda script, or a local file in a pwa studio extension/package (like create-pwa).

I personally prefer keeping this file locally within `create-pwa` and just publishing a new version if the backend changes. It's much more straightforward and requires no knowledge of lambdas, aws routing, etc.

If the team want to keep this solution as is, I'd really like to see some documentation with steps to update this file.

Reply:
The points you have mentioned are perfectly reasonable. I did have a thought of making it a package to avoid the cost of hosting and also avoid potential DoS security issues. But the biggest problem with publishing on NPM is that developers don't generally run `yarn install` every time they build or might be using an older version of the package causing older environments to be used instead. The cost of using AWS Lambda and AWS S3 is minimal and we have enabled throttling as an initial security check.

Reply:
> But the biggest problem with publishing on NPM is that developers don't generally run yarn install every time they build or might be using an older version of the package causing older environments to be used instead.

Two questions: 

1) How often does the `getSampleBackends` service actually need to be used? It should just be used when someone spins up their project for the first time, right?

2) Is the `getSampleBackends` API used during the _build_ or do an up-check to see if the backend is accessible during build?

Reply:
> 1. How often does the `getSampleBackends` service actually need to be used? It should just be used when someone spins up their project for the first time, right?

Along with bootstrapping, It will be used every time someone adds the `venia-sample-backends` extension and builds their project.

> 2. Is the `getSampleBackends` API used during the _build_ or do an up-check to see if the backend is accessible during build?

It is used during the build stage if the extension is installed. It will be used to fetch the sample backends, if the current backend is inactive. We won't be calling the service if the backend mentioned in the ENV file is active.

Reply:
What happens if this URL goes down? Will the yarn install hang for X seconds? 

Also if we're maintaining this micro server we need to setup monitoring to ensure we can be aware of uptime issues. We should also consider any security implications of having this service running.

Reply:
Nice question @davemacaulay. I have used a wrong URL to simulate the service failure and you can see here, if the service is down, the script will use the internal URLs shipped as part of the `sampleBackends.json` file.

![image](https://user-images.githubusercontent.com/35203638/100133533-83e2c080-2e4c-11eb-9f25-fb53128b4c57.png)

You are right about monitoring. I'll create a ticket to investigate the monitoring capabilities of our team/AWS.

Coming to security, we are not giving the public access to the Lamda function or the S3 object. It is an AWS Gateway Service that we are using. Also, the Lamda function that we are using only has read access to that particular object.

Reply:
@revanth0212 It would nice if we can move the lambda setup to aws prod account and if any Audits fail. Rest all looks good.